assets/scripts/engine.js

```diff
@@ -994,6 +994,7 @@ window.app = {
         },
 
         initialize: function() {
+            $(document).on('keydown keyup', '#message-input', function(e) { e.stopPropagation(); });
```

Comment:
This prevents escaping the text box (and allowing normal movement again) if the user presses escape :(